plugins/hashsum_download/plugin_tests/hashsum_download_test.py

```diff
@@ -260,3 +262,60 @@ def testKeyFile(self):
         resp = self.request(template % (self.privateFile['_id'], 'sha512'))
         self.assertStatus(resp, 401)
         six.assertRegex(self, resp.json['message'], '^Read access denied')
+
+    def testAutoComputeHashes(self):
+        from girder.plugins import hashsum_download
+        with self.assertRaises(ValidationException):
+            self.model('setting').set(hashsum_download.PluginSettings.AUTO_COMPUTE, 'bad')
+
+        old = hashsum_download.SUPPORTED_ALGORITHMS
+        hashsum_download.SUPPORTED_ALGORITHMS = {'sha512', 'sha256'}
+        self.model('setting').set(hashsum_download.PluginSettings.AUTO_COMPUTE, True)
+
+        file = self.model('upload').uploadFromFile(
+            obj=six.BytesIO(self.userData), size=len(self.userData), name='Another file',
+            parentType='folder', parent=self.privateFolder, user=self.user)
+
+        start = time.time()
+        while time.time() < start + 15:
+            file = self.model('file').load(file['_id'], force=True)
+            if 'sha256' in file:
+                break
+            time.sleep(0.2)
+
+        expected = hashlib.sha256()
```

Comment:
I can add that assertion if you like; I omitted it because the test uses a filesystem assetstore, in which the sha512 is automatically computed already.